tensorflow/core/kernels/map_stage_op.cc

```diff
@@ -65,6 +65,8 @@ class StagingMap : public ResourceBase
 private:
   // Private variables
   int capacity_;
+  int memory_limit_;
```

Comment:
When dealing with memory sizes, it's probably best to use size_t.  A signed 32-bit integer could easily be exceed.